contractcourt/contract_resolvers.go

```diff
@@ -1315,20 +1315,32 @@ func (c *commitSweepResolver) Resolve() (ContractResolver, error) {
 		log.Infof("%T(%v): sweeping commit output with tx=%v", c,
 			c.chanPoint, spew.Sdump(c.sweepTx))
 
+		// With the sweep transaction constructed, we'll now Checkpoint
+		// our state.
+		if err := c.Checkpoint(c); err != nil {
+			log.Errorf("unable to Checkpoint: %v", err)
+			return nil, err
+		}
+
+		// With the sweep transaction checkpointed, we'll fallthrough to
+		// actually publish the transaction. Upon restart, the resolver
+		// will immediately take the case below since the sweep tx is
+		// checkpointed.
+		fallthrough
```

Comment:
IMO the control flow is easier to follow if we retain the publishing of the tx in this case, and also the ones that need a tx published.